courses-app/src/app/auth/services/auth.service.ts

```diff
@@ -1,117 +1,58 @@
-            this._loading$$.next(false);
-          },
-          error: err => {
-            console.log('Error login at subscribe - auth-service', err)
-            this._resultMessage$$.next(new ResultMessage(false, err))
-            this._loading$$.next(false);
-          }
-        })
-    }
+  constructor(private http: HttpClient) {
   }
 
-  logout(withMessage?: boolean) {
-    this._loading$$.next(true);
-
-    this._authorized$$.next(false);
-    this.userStoreService.deleteUserState();
-    this.router.navigate(['/login'])
-
-    this.http.delete('http://localhost:3000/logout').pipe(
-      finalize(() => {
-        this.sessionStorageService.deleteToken();
-        if (withMessage) {
-          this._resultMessage$$.next(new ResultMessage(true, ['Logout successful']))
-        }
-        this._loading$$.next(false);
-      })
-    ).subscribe()
-  }
-
-  register(user: { name: string, email: string, password: string }): void {
-    this._loading$$.next(true);
-
-    // @TODO refactor to this ->
-    // this.userService.registerUser(user).subscribe((result: ResultMessage) => {
-    //   this._resultMessage$$.next(new ResultMessage(result.successful, result.messages))
-    //   this._loading$$.next(false);
-    // })
-
-    this.userService.registerUser(user).subscribe({
-      next: success => {
-        console.log(success)
-        this._resultMessage$$.next(new ResultMessage(true, [success]))
-        this._loading$$.next(false);
-      },
-      error: err => {
-        console.log(err)
-        this._resultMessage$$.next(new ResultMessage(false, err.errors))
-        this._loading$$.next(false);
-      }
-    })
+  public login(user: { email: string, password: string }): Observable<string> {
+    return this.http.post<ISignInResponse>('http://localhost:3000/login', user)
+      .pipe(
+        map((result: ISignInResponse) => {
+          return result.result;
+        }),
+        catchError((error: HttpErrorResponse) => {
+          return throwError([error.error.error])
+        }),
+      )
   }
 
-  private initUserSession() {
-    let token: string | null = sessionStorage.getItem('token');
-    if (token) {
-      this.userStoreService.setUserSession().subscribe({
-        next: () => {
-          this._authorized$$.next(true);
-        },
-        error: (error) => {
-          console.log(error)
-          this._resultMessage$$.next(new ResultMessage(false, error.errors))
-        },
-      })
-    }
+  public register(user: { name: string, email: string, password: string }): Observable<string> {
+    return this.http
+      .post<{ successful: boolean, result: string }>('http://localhost:3000/register', user)
+      .pipe(
+        catchError((error: HttpErrorResponse) => {
+          console.log('pure error: ', error.error)
+          return throwError(error.error)
+        }),
+        map(((next: any) => {
+          console.log('pure next: ', next)
```

Comment:
as something extra, try to write an operator for logging: https://angular-schule.github.io/website-articles/blog/2018-02-rxjs-own-log-operator/README.html , https://medium.com/hackernoon/the-ultimate-rxjs-operator-fae6414d5db4